static/js/narrow.js

```diff
@@ -269,17 +270,23 @@ export function activate(raw_operators, opts) {
     save_pre_narrow_offset_for_reload();
 
     let msg_data = new MessageListData({
-        filter: narrow_state.filter(),
+        filter,
         excludes_muted_topics,
     });
 
+    const mld_exists = mld_cache.has(filter);
+    if (mld_exists) {
+        msg_data = mld_cache.get(filter);
+    }
```

Comment:
This should probably be an if/else situation, where we don't call `new MessageListData` at all if we're using the cache?  Not sure, because of the discussion below.